### PR TITLE
fix: Ensure SQL streams are sorted when a replication key is set

### PR DESCRIPTION
**Pull request by edgarrmondragon**
----
:books: Documentation preview :books:: https://meltano-sdk--1951.org.readthedocs.build/en/1951/

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
SQLAlchemy 1.* tests are failing because inspected columns have a different order, so the SCHEMA messages, and thus the snapshots, are different.

File: tests/samples/test_target_sqlite.py [hunk 1]
```diff
@@ -84,8 +85,10 @@ def test_sync_sqlite_to_sqlite(
     )
     orig_stdout.seek(0)
     tapped_config = dict(sqlite_sample_target.config)
-    catalog = _get_tap_catalog(SQLiteTap, config=tapped_config, select_all=True)
-    tapped_target = SQLiteTap(config=tapped_config, catalog=catalog)
+    tapped_target = SQLiteTap(
+        config=tapped_config,
+        catalog=sqlite_sample_db_catalog.to_dict(),
+    )
```
Comment (edgarrmondragon) [inline tests/samples/test_target_sqlite.py:91]: The data tapped from the loaded SQLite db had a different catalog, i.e. streams that are incremental during the first sync, are "full table" in the downstream sync, so the message count was different and the test was failing.

File: tests/samples/test_tap_sqlite.py [hunk 2]
```diff
@@ -116,3 +119,24 @@ def test_sync_sqlite_to_csv(sqlite_sample_tap: SQLTap, tmp_path: Path):
...
+            state_messages.append(message)
+
+    return state_messages
+
+
+def test_sqlite_state(sqlite_sample_tap_state_messages):
+    assert all(
+        "progress_markers" not in bookmark
+        for message in sqlite_sample_tap_state_messages
+        for bookmark in message["value"]["bookmarks"].values()
```
Comment (edgarrmondragon) [inline tests/samples/test_tap_sqlite.py:141]: This ensures that the stream is resumable

**Review by pnadolny13 — APPROVED** [review]
@edgarrmondragon this looks good to me!

**Review by kgpayne — APPROVED** [review]

**Pull request merged**